File: src/main/java/me/santio/uhc/events/PlayerRevivedEvent.java
package me.santio.uhc.events;

import lombok.Getter;
import lombok.Setter;
import me.santio.uhc.models.GamePlayer;
import org.bukkit.event.Cancellable;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;

public class PlayerRevivedEvent extends Event implements Cancellable {
    
    @Setter @Getter private boolean cancelled = false;
    @Getter private final GamePlayer player;
    @Getter private final GamePlayer admin;
    private final HandlerList HANDLER_LIST = new HandlerList();
    
    public PlayerRevivedEvent(GamePlayer player, GamePlayer admin) {
        this.player = player;
        this.admin = admin;
    }
    
    @Override
    public HandlerList getHandlers() {
        return HANDLER_LIST;
    }
}

File: src/main/kotlin/me/santio/uhc/events/PlayerRevivedEvent.kt
package me.santio.uhc.events

import me.santio.uhc.models.GamePlayer
import org.bukkit.event.Cancellable
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

class PlayerRevivedEvent(private val player: GamePlayer, private val admin: GamePlayer) : Event(), Cancellable {
    private var cancelled = false
    private val HANDLER_LIST = HandlerList()

    override fun getHandlers(): HandlerList = HANDLER_LIST
    override fun isCancelled(): Boolean = cancelled

    override fun setCancelled(cancel: Boolean) {
        cancelled = cancel
    }
}